packages/message/src/Granules.ts

```diff
@@ -133,13 +135,15 @@ function isProcessingTimeInfo(
 
 /**
  * ** Private **
- * Convert date string to standard ISO format, retaining null values if they exist
+ * Convert date string to standard ISO format, retaining null/undefined values if they exist
+ * and converting '' to null
  *
  * @param {string} date - Date string, possibly in multiple formats
  * @returns {string} Standardized ISO date string
  */
-const convertDateToISOStringPreservingNull = (date: string | null) => {
-  if (date === null) return null;
+const convertDateToISOStringSettingNull = (date: string | null | undefined) => {
+  if (isNil(date)) return date;
+  if (date === '') return null;
```

Comment:
I think this is good and is more explicit than something like `if (!date)` but is there a case here where just checking for a falsy value with `!date` falls down?

Reply:
Not really, no, for the set `(date: string | null | undefined) `  L145 handles `null | undefined`, and (!date) would handle the string case, however it felt like in this case it obscured the intent/reduced explicitness in favor of a blanket implicit behavior. 

tl;dr favoring explicit intent over implicit coverage here, it felt more defensive against unintended future changes.